src/lib.rs

```diff
@@ -1941,4 +1941,9 @@ pub struct Module {
     pub functions: Arena<Function>,
     /// Entry points.
     pub entry_points: Vec<EntryPoint>,
+
+    /// Source Code for debug
+    pub source_code: Option<String>,
+    /// File name for debug
+    pub file_path: Option<String>,
```

Comment:
Although this pr is for spv, I think this can be useful if we do the debug info for other backends.
For backends that generate source code, we can track the original file and line number, then insert this information as comments.
Metal also has a debug symbol file format that might work like SPV, but the format has no document at all (I’m not sure if it’s a DWARF or Apple’s proprietary format).

Reply:
The idea was that every backend that supports this would take this info via their write functions, we wouldn't have to clone the source either (having the source have a lifetime in the module doesn't seem worth it).

> Metal also has a debug symbol file format that might work like SPV, but the format has no document at all (I’m not sure if it’s a DWARF or Apple’s proprietary format).

Are you referring to [metallibsym](https://developer.apple.com/documentation/metal/shader_libraries/generating_and_loading_a_metal_library_symbol_file?language=objc)?

We have some info about [metallib](https://github.com/gfx-rs/naga/wiki/Metallib-file-format) ([docs](https://developer.apple.com/documentation/metal/shader_libraries/compiling_shader_code_into_a_library_with_metal_s_command-line_tools?changes=latest_minor%20rel%3D&language=objc)) but not about the symbols file.